diff --git a/aosp_source/development/tools/make_key b/aosp_source/development/tools/make_key
index 48224e9..591295a 100644
--- a/aosp_source/development/tools/make_key
+++ b/aosp_source/development/tools/make_key
@@ -46,8 +46,12 @@ mknod ${one} p
 mknod ${two} p
 chmod 0600 ${one} ${two}
 
-read -p "Enter password for '$1' (blank for none; password will be visible): " \
-  password
+# Use password from $password environment variable
+password="$password"
+
+if [ -z "$password" ]; then
+  read -p "Enter password for '$1' (blank for none; password will be visible): " password
+fi
 
 if [ "${3}" = "rsa" -o "$#" -eq 2 ]; then
   ( openssl genrsa -f4 2048 | tee ${one} > ${two} ) &
